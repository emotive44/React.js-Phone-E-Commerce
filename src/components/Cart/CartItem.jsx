import React from 'react';

export default function CartItem({ item, data }) {
    if(!item || !data) { return null }
    const { id, title, img, price, total, count } = item;
    const { increment, decrement, removeItem } = data;
    return (
        <div className="row text-capitalize text-center my-2">
            <div className="col-10 mx-auto col-lg-2">
                <img src={img} alt="product" style={{width: '5rem', height:'5rem'}} className="img-fluid"/>
            </div>
            <div className="col-10 mx-auto col-lg-2 my-3">
                <span className="d-lg-none">Product : </span>
                {title}
            </div>
            <div className="col-10 mx-auto col-lg-2 my-3">
                <span className="d-lg-none">Price : </span>
                {price}
            </div>
            <div className="col-10 mx-auto col-lg-2 my-3 y-lg-0">
                <div className="d-flex justify-content-center">
                    <span className="btn btn-black mx-1" onClick={() => decrement(id)}>-</span>
                    <span className="btn btn-black mx-1">{count}</span>
                    <span className="btn btn-black mx-1" onClick={() => increment(id)}>+</span>
                </div>
            </div>
            <div className="col-10 mx-auto col-lg-2 my-3">
                <div className="cart-icon" onClick={() => removeItem(id)}>
                    <i className="fas fa-trash"/>
                </div>
            </div>
            <div className="col-10 mx-auto col-lg-2 my-3">
                <strong>Item Total : ${total}</strong>
            </div>
        </div>
    )
}