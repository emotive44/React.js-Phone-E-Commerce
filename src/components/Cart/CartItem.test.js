import React from 'react';
import CartItem from './CartItem';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('CartItem component', () => {
    test('should reder null if item or data are undefined', () => {
        const component = renderer.create(<CartItem />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})