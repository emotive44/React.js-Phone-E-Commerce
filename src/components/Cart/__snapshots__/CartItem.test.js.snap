// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component should reder null if item or data are undefined 1`] = `null`;